### PR TITLE
v1.3.5: more descriptive error handling and examples is now marked as optional

### PR DESCRIPTION
**Pull request by choicallum**

### CODE REVIEW

**Comment by choicallum** [comment]
I have no clue why these tests run for forever, going to take a look at this later.

**Comment by choicallum** [comment]
I have no clue why these tests run for forever, going to take a look at this later.

File: cohere/assert_parameters.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+def get_base_error (expected_type: Union[type, tuple], name: str) -> str:
+    base_error = ""
+    if hasattr(expected_type, '__name__'):
+        return f"the {name} parameter is expected to be a list of {expected_type.__name__},"
+    else:
+        expected_types = expected_type[0].__name__
+        for my_type in expected_type[1:]:
+            expected_types += " or a " + my_type.__name__ 
+        return f"the {name} parameter is expected to be a {expected_types},"
```
Comment (OrenLeung) [inline cohere/assert_parameters.py:40]: test cases

Comment (OrenLeung) [inline cohere/assert_parameters.py:40]: ty for taking the intiative to do runtime type checking!

have you thought about using a decorator pattern for this similar to this but with cleaner error msg?

https://github.com/agronholm/typeguard

Comment (choicallum) [reply cohere/assert_parameters.py:40]: hm... while interesting it's not rly something I'm willing to spend time to understand and implement currently w/ my lack of python knowledge, maybe if i have some more time later, resolving for now.

**Pull request closed without merging**